staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -100,13 +93,32 @@ func ParseEncryptionConfiguration(f io.Reader) (map[schema.GroupResource]value.T
 		result[gr] = value.NewMutableTransformer(value.NewPrefixTransformers(fmt.Errorf("no matching prefix found"), transList...))
 	}
 	return result, nil
+
+}
+
+// loadConfig decodes data as a EncryptionConfiguration object.
+func loadConfig(data []byte) (*cfg.EncryptionConfiguration, error) {
+	scheme := runtime.NewScheme()
+	codecs := serializer.NewCodecFactory(scheme)
+	cfg.AddToScheme(scheme)
+	cfgv1.AddToScheme(scheme)
+
+	configObj, gvk, err := codecs.UniversalDecoder().Decode(data, nil, nil)
```

Comment:
Do we have a better way for parsing YAML? I couldn't find a way of getting a scheme.

CC @sttts

Reply:
This is what we do for the audit API policy file https://github.com/kubernetes/kubernetes/blob/484f62a5686f718429907c2fec8406749e2f6260/staging/src/k8s.io/apiserver/pkg/audit/policy/reader.go#L47